arkouda/strings.py

```diff
@@ -0,0 +1,305 @@
+                                                              other)
+        else:
+            raise ValueError("Strings: {} not supported between Strings and {}".format(op, type(other)))
+        repMsg = generic_msg(msg)
+        return create_pdarray(repMsg)
+        
+    def __eq__(self, other):
+        return self.binop(other, "==")
+
+    def __neq__(self, other):
+        return self.binop(other, "!=")
+
+    def __getitem__(self, key):
+        if np.isscalar(key) and resolve_scalar_dtype(key) == 'int64':
+            if (key >= 0 and key < self.size):
+                msg = "segmentedIndex {} {} {} {} {}".format('intIndex',
+                                                             self.objtype,
+                                                             self.offsets.name,
+                                                             self.bytes.name,
+                                                             key)
+                repMsg = generic_msg(msg)
+                fields = repMsg.split()
+                # value = fields[2]
+                return parse_single_value(' '.join(fields[1:]))
+            else:
+                raise IndexError("[int] {} is out of bounds with size {}".format(key,self.size))
+        elif isinstance(key, slice):
+            (start,stop,stride) = key.indices(self.size)
+            if verbose: print(start,stop,stride)
+            msg = "segmentedIndex {} {} {} {} {} {} {}".format('sliceIndex',
+                                                               self.objtype,
+                                                               self.offsets.name,
+                                                               self.bytes.name,
+                                                               start,
+                                                               stop,
+                                                               stride)
+            repMsg = generic_msg(msg)
+            offsets, values = repMsg.split('+')
+            return Strings(offsets, values);
+        elif isinstance(key, pdarray):
+            kind, itemsize = translate_np_dtype(key.dtype)
+            if kind not in ("bool", "int"):
+                raise TypeError("unsupported pdarray index type {}".format(key.dtype))
+            if kind == "bool" and self.size != key.size:
+                raise ValueError("size mismatch {} {}".format(self.size,key.size))
+            msg = "segmentedIndex {} {} {} {} {}".format('pdarrayIndex',
+                                                         self.objtype,
+                                                         self.offsets.name,
+                                                         self.bytes.name,
+                                                         key.name)
+            repMsg = generic_msg(msg)
+            offsets, values = repMsg.split('+')
+            return Strings(offsets, values)
+        else:
+            return NotImplemented
```

Comment:
Should this be a raise key type error?

Reply:
Yes, I'll change it to raise a TypeError